lib/kennel/syncer.rb

```diff
@@ -254,11 +254,17 @@ def prevent_irreversible_partial_updates
     end
 
     def populate_id_map(expected, actual)
+      expected.each do |e|
+        @id_map.add_new(e.class.api_resource, e.tracking_id)
+      end
```

Comment:
this seems wrong, not everything that is "expected" is new

Reply:
True, but it's immediately overwritten by the next section.

Reply:
ah true, the old code did the sane but more complicated 👍